src/core/service.c

```diff
@@ -2744,10 +2744,8 @@ static int service_deserialize_item(Unit *u, const char *key, const char *value,
                 r = cunescape(value, 0, &t);
                 if (r < 0)
                         log_unit_debug_errno(u, r, "Failed to unescape status text: %s", value);
-                else {
-                        free(s->status_text);
-                        s->status_text = t;
-                }
+                else
+                        free_and_replace(s->status_text, t);
```

Comment:
This is slightly off-topic, but this bothered me while giving this a casual look:

Why is `free_and_replace()` a macro?  To a casual reader not looking at the definition, this usage looks incorrect, and I see no obvious reason for this not to be a simple function like:

```
void free_and_replace(void **ptr, void *replacement)
{
        assert(ptr);

        free(*ptr);
        *ptr = replacement;
}
```

Then the call sites wouldn't look so anomalous: `free_and_replace(&s->status_text, t)`

I'm sure there's a reason, but I don't know what it is, and there's no comment explaining it in `basic/alloc-util.h`.

Reply:
it does actually do two more things, it also resets *replacement and returns NULL. This means that the function would have to take two ptrs to ptrs actually, but iirc if you do that then we don't get the free implicit casts from any ptr type anymore, i.e. every ptr will implicitly cast to void*, but not any ptr of a ptr to void**, iirc

Reply:
That's fair, but maybe these `free_and_replace()` and `strv_free_and_replace()` macros should be capitalized to make it obvious to the reader that they're macros and not ostensibly strange function calls to the casual reader.

/bikeshedding